File: tests/integration-tests/src/test/java/com/jaynewstrom/json/sample/simple/SkiResortTest.java
package com.jaynewstrom.json.sample.simple;

import com.jaynewstrom.json.sample.JsonTestHelper;

import org.junit.Test;

import static org.fest.assertions.api.Assertions.assertThat;

public final class SkiResortTest {
    @Test public void ensureSkiResortIsDeserializedFromJson() {
        SkiResort skiResort = new JsonTestHelper().deserializeFile(SkiResort.class, "JacksonHole.json", this);
        assertThat(skiResort.name).isEqualTo("Jackson Hole");
        assertThat(skiResort.population).isEqualTo(10_135);
        assertThat(skiResort.averageSnowfallInInches).isEqualTo(459);
    }

    @Test public void ensureSkiResortIsCreatedFromJson() {
        SkiResort telluride = new SkiResort("Telluride", 2_319, 309);
        String json = new JsonTestHelper().serialize(telluride);
        assertThat(json).isEqualTo("{\"name\":\"Telluride\",\"population\":2319,\"averageSnowfall\":309}");
    }
}

File: tests/integration-tests/src/test/java/com/jaynewstrom/json/sample/simple/SkiResortTest.kt
package com.jaynewstrom.json.sample.simple

import com.jaynewstrom.json.sample.JsonTestHelper
import org.fest.assertions.api.Assertions.assertThat
import org.junit.Test

class SkiResortTest {
    @Test fun ensureSkiResortIsDeserializedFromJson() {
        val (name, population, averageSnowfallInInches) = JsonTestHelper().deserializeFile(SkiResort::class.java, "JacksonHole.json", this)
        assertThat(name).isEqualTo("Jackson Hole")
        assertThat(population).isEqualTo(10135)
        assertThat(averageSnowfallInInches).isEqualTo(459)
    }

    @Test fun ensureSkiResortIsCreatedFromJson() {
        val telluride = SkiResort("Telluride", 2319, 309)
        val json = JsonTestHelper().serialize(telluride, SkiResort::class.java)
        assertThat(json).isEqualTo("{\"name\":\"Telluride\",\"population\":2319,\"averageSnowfall\":309}")
    }
}
